app/src/main/java/com/mediapipe/example/sign_language/HolisticLandmarkerHelper.kt

```diff
@@ -0,0 +1,189 @@
+    val minFaceDetectionConfidence: Float,
+    val minPoseDetectionConfidence: Float,
+    val minFaceSuppressionThreshold: Float,
+    val minPoseSuppressionThreshold: Float,
+    // this listener is only used when running in RunningMode.LIVE_STREAM
+    private val landmarkerHelperListener: LandmarkerListener? = null
+) {
+
+    companion object {
+        private const val MP_HOLISTIC_LANDMARKER_TASK =
+            "holistic_landmarker.task"
+        const val TAG = "HolisticLandmarkerHelper"
+        const val OTHER_ERROR = 0
+        const val DEFAULT_MIN_FACE_PRESENCE_CONFIDENCE = 0.5F
+        const val DEFAULT_MIN_HAND_LANDMARKS_CONFIDENCE = 0.5F
+        const val DEFAULT_MIN_POSE_PRESENCE_CONFIDENCE = 0F
+        const val DEFAULT_MIN_FACE_DETECTION_CONFIDENCE = 0.5F
+        const val DEFAULT_MIN_POSE_DETECTION_CONFIDENCE = 0F
+        const val DEFAULT_MIN_FACE_SUPPRESSION_THRESHOLD = 0.5F
+        const val DEFAULT_MIN_POSE_SUPPRESSION_THRESHOLD = 0F
+
+        const val MIN_CONFIDENCE = 0F
+        const val MAX_CONFIDENCE = 1F
+    }
+
+    private var holisticLandmarker: HolisticLandmarker? = null
+
+    init {
+        setUpHolisticLandmarker()
+    }
+
+    private fun setUpHolisticLandmarker() {
+        val baseOptionBuilder = BaseOptions.builder()
+
+        baseOptionBuilder.setDelegate(Delegate.CPU)
+
+        // Check if runningMode is consistent with landmarkerHelperListener
+        when (runningMode) {
+            RunningMode.LIVE_STREAM -> {
+                if (landmarkerHelperListener == null) {
+                    throw IllegalStateException(
+                        "holisticLandmarkerHelperListener must be set when runningMode is LIVE_STREAM."
+                    )
+                }
+            }
+
+            else -> {
+                // no-op
+            }
+        }
+        try {
+            baseOptionBuilder.setModelAssetPath(MP_HOLISTIC_LANDMARKER_TASK)
+            val baseOptions = baseOptionBuilder.build()
+            val optionsBuilder =
+                HolisticLandmarker.HolisticLandmarkerOptions.builder()
+                    .setBaseOptions(baseOptions).setRunningMode(runningMode)
+                    .setMinHandLandmarksConfidence(minHandLandmarksConfidence)
+                    .setMinFaceDetectionConfidence(minFaceDetectionConfidence)
+                    .setMinPoseDetectionConfidence(minPoseDetectionConfidence)
+                    .setMinFaceSuppressionThreshold(minFaceSuppressionThreshold)
+                    .setMinPoseSuppressionThreshold(minPoseSuppressionThreshold)
+                    .setMinFacePresenceConfidence(minFacePresenceConfidence)
+                    .setMinPosePresenceConfidence(minPosePresenceConfidence)
+
+            // The ResultListener and ErrorListener only use for LIVE_STREAM mode.
+            if (runningMode == RunningMode.LIVE_STREAM) {
+                optionsBuilder
+                    .setResultListener(this::returnLivestreamResult)
```

Comment:
As we move forward with other migrations, we'll likely update all of our samples to use flow/compose/modern best practices for Android. If you have time now, it could be worthwhile to start that now with this project.

Reply:
@PaulTR I want to confirm
Will we prioritize this project over `Audio Classifier`? If so, I will start migrating it now